components/pages/privacy/access/requests/index.jsx

```diff
@@ -39,35 +43,36 @@ import {
   TOS_PREDICATE,
 } from "../../../../../__testUtils/mockApp";
 import { getRequestorWebId } from "../../../../../src/models/consent/request";
-import { getProfileResource } from "../../../../../src/solidClientHelpers/resource";
 import Spinner from "../../../../spinner";
 
 const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 
 export default function ConsentShow() {
   const { session } = useSession();
   const { fetch } = session;
-  const router = useRouter();
-  const { requestVc } = router.query;
+  const redirectUrl = window.location.href;
   const bem = useBem(useStyles());
   const [consentRequest, setConsentRequest] = useState(null);
   const agentWebId = getRequestorWebId(consentRequest);
   const [agentDetails, setAgentDetails] = useState({});
   const { agentName, agentUrl, agentPolicy, agentTOS } = agentDetails || null;
 
   useEffect(() => {
-    if (!requestVc) return;
-    const req = JSON.parse(atob(requestVc));
-    setConsentRequest(req);
-  }, [requestVc, fetch]);
+    (async () => {
+      if (!redirectUrl) return;
+      const { accessRequest } = await getAccessRequestFromRedirectUrl(
+        redirectUrl,
+        { fetch }
+      );
+      setConsentRequest(accessRequest);
+    })();
+  }, [redirectUrl, fetch]);
 
   useEffect(() => {
     if (!consentRequest) return;
     (async () => {
-      const { dataset: profileDataset } = await getProfileResource(
-        agentWebId,
-        fetch
-      );
+      const profiles = await getProfileAll(agentWebId);
+      const profileDataset = profiles.webIdProfile || profiles.altProfileAll[0]; // getting the first profile for now
```

Comment:
you could also do just `profiles.altProfileAll.first((profile) => !!profile)`, that is "find the first alternative profile which could be loaded"